RELEASES.md

```diff
@@ -0,0 +1,248 @@
+# Versioning and Release
+
+This document details the versioning and release plan for containerd. Stability
+is a top goal for this project and we hope that this document and the processes
+it entails will help to achieve that. It covers the release process, versioning
+numbering, backporting, API stability and support horizons.
+
+If you rely on containerd, it would be good to spend time understanding the
+areas of the API that are and are not supported and how they impact your
+project in the future.
+
+This document will be considered a living document. Supported timelines,
+backport targets and API stability guarantees will be updated here as they
+change.
+
+If there is something that you require or this document leaves out, please
+reach out by [filing an issue](https://github.com/containerd/containerd/issues).
+
+## Releases
+
+Releases of containerd will be versioned using dotted triples, similar to
+[Semantic Version](http://semver.org/). For the purposes of this document, we
+will refer to the respective components of this triple as
+`<major>.<minor>.<patch>`. The version number may have additional information,
+such as alpha, beta and release candidate qualifications. Such releases will be
+considered "pre-releases".
+
+### Major and Minor Releases
+
+Major and minor releases of containerd will be made from master. Releases of
+containerd will be marked with GPG signed tags and announced at
+https://github.com/containerd/containerd/releases. The tag will be of the
+format `v<major>.<minor>.<patch>` and should be made with the command `git tag
+-s v<major>.<minor>.<patch>`.
+
+After a minor release, a branch will be created, with the format
+`release/<major>.<minor>` from the minor tag. All further patch releases will
+be done from that branch. For example, once we release `v1.0.0`, a branch
+`release/1.0` will be created from that tag. All future patch releases will be
+done against that branch.
+
+### Pre-releases
+
+Pre-releases, such as alphas, betas and release candidates will be conducted
+from their source branch. For major and minor releases, these releases will be
+done from master. For patch releases, these pre-releases should be done with in
+the corresponding release branch.
+
+While pre-releases are done to assist in the stabilization process, no
+guarantees are provided.
+
+### Upgrade Path
+
+The upgrade path for containerd is such that the 0.0.x patch releases are
+always backward compatible with its major and minor version. Minor (0.x.0)
+version will always be compatible with the previous minor release. i.e. 1.2.0
+is backwards compatible with 1.1.0 and 1.1.0 is compatible with 1.0.0. There is
+no compatibility guarantees with upgrades from two minor releases. i.e. 1.0.0
+to 1.2.0.
```

Comment:
The phrase "from two minor releases" is difficult to grok I think perhaps "upgrades spanning multiple minor releases"?.

Also I think it's either "There are no compatibility guarantees" or "There is no compatibility guarantee" rather than "There is no compatibility guarantees".